tools/biosplitmerge.c

```diff
@@ -0,0 +1,192 @@
+/* 
+ * biosplitmerge.c      The eBPF C source file for split and merge tracer. 
+ *                      Functions in this file are imported and used by 
+ *                      biosplitmerge.py. 
+ *
+ *
+ * Copyright (c) Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * 09-Aug-2020   Haoning Chen   Created this.
+ */
+
+#include <uapi/linux/ptrace.h>
+#include <linux/blkdev.h>
+#include <linux/version.h>
+
+/* Split/Merge event type. */
+enum type { Split = 0, Fmerge, Bmerge, Dmerge };
+
+
+/* Value of the bpf map */
+struct val_t {
+	u64 ts; 
+	u64 sector;
+	u64 len;
+	u64 type;
+};
+
+
+/* Output data type sent from kernel to user space */
+struct data_t {
+	u64 ts; 
+	u64 in_sector;
+	u64 in_len;
+	u64 out_sector; 
+	u64 out_len; 
+	u64 rwflag; 
```

Comment:
does this need to be u64 if just bool?

Reply:
I shrank the size of the structs by setting rwflag to u8, split/merge type to u8 and pid to u32. 

It turns out that there's the u64-alligned requirement for bcc's struct if the struct is initialized using compound initialization (e.g, struct data_t data = { .a = ... } ). But if the struct's data gets filled one by one, it is ok. So I removed compound assignment when initializing the structs.